airbyte-integrations/connectors/source-stripe/source_stripe/partition.py

```diff
@@ -0,0 +1,130 @@
+        request_headers: Optional[Mapping[str, Any]] = None,
+        request_params: Optional[Mapping[str, Any]] = None,
+        request_body_data: Optional[Union[Mapping[str, Any], str]] = None,
+        request_body_json: Optional[Mapping[str, Any]] = None,
+        log_formatter: Optional[Callable[[requests.Response], Any]] = None,
+    ) -> Iterable[Mapping]:
+        next_page_token = None
+        pagination_complete = False
+        while not pagination_complete:
+            if next_page_token:
+                next_page_parameter = {self._page_token_option.field_name: next_page_token["next_page_token"]}
+
+                if self._page_token_option.inject_into == RequestOptionType.request_parameter:
+                    request_params = request_params | next_page_parameter
+                elif self._page_token_option.inject_into == RequestOptionType.header:
+                    request_headers = request_headers | next_page_parameter
+                elif self._page_token_option.inject_into == RequestOptionType.body_data:
+                    request_body_data = request_body_data | next_page_parameter
+                elif self._page_token_option.inject_into == RequestOptionType.body_json:
+                    request_body_json = request_body_json | next_page_parameter
+
+            response = self._requester.send_request(
+                path=path,
+                request_headers=request_headers,
+                request_params=request_params,
+                request_body_data=request_body_data,
+                request_body_json=request_body_json,
+                next_page_token=next_page_token,
+                log_formatter=log_formatter,
+            )
+            if not response:
+                pagination_complete = True
+            else:
+                records = list(self._record_selector.select_records(response, {}))
+                yield from records
+
+                next_page_token = self._paginator.next_page_token(response, records)
+                if not next_page_token:
+                    pagination_complete = True
+
+
+class StripePartitionGenerator(PartitionGenerator):
+    def __init__(self, stream: Stream, message_repository: MessageRepository, requester: PaginatedRequester):
+        self._stream = stream
+        self._message_repository = message_repository
+        self._requester = requester
+
+    def generate(self, sync_mode: SyncMode) -> Iterable[Partition]:
+        for s in self._stream.stream_slices(sync_mode=sync_mode):
+            yield StripePartition(copy.deepcopy(s), self._message_repository, self._requester)
+
+
+class SourcePartition(Partition):
+    def __init__(self, _slice: Mapping[str, Any], message_repository: MessageRepository, requester: PaginatedRequester):
+        self._slice = _slice
+        self._message_repository = message_repository
+        self._requester = requester
+
+    @property
+    def request_parameters(self) -> Optional[Mapping[str, Any]]:
```

Comment:
we shouldn't expose these properties publicly for arbitrary source because not all sources are HTTP API sources